### DISCUSSION TITLE
Chat Embed

### DISCUSSION THREAD

**Discussion by amansoniamazatic** [FlowiseAI/Flowise#1773] [Ideas]
Hii @HenryHengZJ @Jaredude , I am currently facing a situation where I have embedded the chatbot in a production website. Now, I want the chat window to refresh after a specific time or after ten messages in a conversation. Additionally, when a user leaves the page and returns, I want the conversation to start from the beginning instead of continuing from where it left off and removed the previous conversation is this possible can you give me any idea.

**Comment by Jaredude** [comment]
I would fork the separate [chatembed repo](https://github.com/FlowiseAI/FlowiseChatEmbed) and make the enhancements to have a fully custom chatembed with the features you're talking about. That provides you with the ability to have an embed that works exactly as you built it.

A feature in the existing chatembed that will help you accomplish much of what you're looking to do is the observable config. You can checkout the documentation on the chatembed to get a basic understanding of how it works. In a nutshell, every message sent and received is observable in callback functions in the config of the chat init. This means you can then use local state/variables to keep track of the messages as they come and go and perform logic in the parent container/window based upon whatever logic you want. Hope that helps.

**Reply by L88iO9n5Gs** [reply]
hey @Jaredude, one general question: I forked the repo and wanted to make some changes. after uncommenting server and livereload variables in rollup.config.js, how can i then see anything? When I try yarn dev, a 404 page opens.

**Comment by Jaredude** [comment]
As far as resetting the previous conversation if the user leaves the page and returns, what you're talking about there is session management that you'd need to manage on the web server rather than flowise. What you'd do there is manage a session variable on your website in a way that will ensure that each time the user returns to that page with the bot, it will then use overrideConfig to replace the sessionId value with something that it unique based upon whatever business rules you define (e.g. new every time the user arrives on the page; back button, refresh, 10 questions, etc. whatever logic you define on the website).

**Reply by amansoniamazatic** [reply]
Thanks @Jaredude
